### DISCUSSION TITLE
Typo in docs?

### DISCUSSION THREAD

**Discussion by HeartlandTechie** [filamentphp/filament#1709] [General]
https://filamentphp.com/docs/2.x/tables/actions#modals

Should that be model instead of modal?

`use App\Modals\User;
use Filament\Forms;
use Filament\Tables\Actions\BulkAction;
use Illuminate\Database\Eloquent\Collection;
 
BulkAction::make('updateAuthor')
    ->action(function (Collection $records, array $data): void {
        foreach ($records as $record) {
            $record->author()->associate($data['authorId']);
            $record->save();
        }
    })`

**Comment by danharrin** [comment]
No, because actions that require confirmation / have forms associated with them open modals! :)

**Reply by HeartlandTechie** [reply]
I want to see how that works as a complete running piece of code.

**Reply by danharrin** [reply]
Used a text input here for simplicity but you can replace it with a Select if you wish

```php
use App\Modals\User;
use Filament\Forms;
use Filament\Tables\Actions\BulkAction;
use Illuminate\Database\Eloquent\Collection;

BulkAction::make('updateAuthor')
    ->form([TextInput::make('authorId')->required()])
    ->action(function (Collection $records, array $data): void {
        foreach ($records as $record) {
            $record->author()->associate($data['authorId']);
            $record->save();
        }
    })
```

Note: not tested this, but you shouldnt have issues

**Reply by HeartlandTechie** [reply]
Let me see the use App\Modals\User;

**Reply by danharrin** [reply]
Whoops, typo. Very sorry, I misunderstood you. That should be `Models`. I've fixed it in the docs.
